### ISSUE TITLE
Port to multiple platforms

### ISSUE THREAD

**Issue by eteran** [eteran/edb-debugger#261]
edb is currently in a state where Linux is the only platform where it even compiles. I'd like to improve this situation, but also don't want to have an issue for each platform like we've had in the past. So this issue will be used to track porting progress instead:

- [ ] Port to FreeBSD
- [ ] Port to OpenBSD
- [ ] Port to OSX
- [ ] Port to Windows

99% of the effort will be in getting the `DebuggerCore` plugins up to speed on those platforms. That's where essentially all of the OS specific code resides.

**Comment by 10110111** [comment]
It seems to me that we'll want `DebuggerCore` to have more arch-specific interface, otherwise we'll end up duplicating much of the arch-specific things like storage for registers, corresponding getters etc..

**Comment by eteran** [comment]
Right, some of things like the `PlatformState` have a new concept of being "filled" from native structures, and are more generic than they used to be. So it can probably be shared between all x86 builds of the plugin. I suspect similar for some of the other `Platform*` classes

**Comment by AaronOpfer** [comment]
Is there a definitive way to set up a reasonable build environment for Windows? I downloaded the Windows 10 Evaluation VM and I'm not sure how to set it up to start building edb. I've installed cmake, VS2017, boost and pkg-config, but I appear to be stuck on getting capstone recognized now. In addition, getting pkg-config recognized involved dropping binaries from http://stackoverflow.com/a/1711338 into C:\Windows, so it seems like there must be a better way.

**Comment by eteran** [comment]
It has been quite a bit since edb built on Windows. Large architectural changes have occurred and new dependencies (capstone) have been introduced.

So, I'll have to look into how to get things going on Windows.

**Comment by AaronOpfer** [comment]
Hey @eteran, did you ever come up with anything for working with Windows? Times like these I wish I understood CMake at all so I could fix it myself.

**Comment by eteran** [comment]
@AaronOpfer unfortunately, I never quite got around to trying to get the windows build system functional. In the end it shouldn't be **too** hard. But it certainly won't succeed in building as the `DebuggerCore` plugin has a wildly different API at this point compared to when the windows port was first attempted.

I do plan to get around to it eventually, but in all honesty, I am prioritizing cleaning up the code and reworking some things to make ARM support doable.

**Comment by eteran** [comment]
@AaronOpfer , @mistersandman  has made some decent headway towards getting things at least building on windows. It's not there yet, but it is **so** much closer than it was. I haven't tried it yet, but if we look at the appveyor script for testing win32 builds, we'll probably get a decent idea of how we can set up functional dev environments for windows.

**Comment by AaronOpfer** [comment]
Cool, I might take a look next time I need a pet project. Don't let me stop
anyone else from diving in, though.
